File: wagerproof-ios-native/Wagerproof/Features/NBA/Components/NBARecentTrendsWidget.swift
import SwiftUI
import WagerproofDesign
import WagerproofModels

/// NBA recent-trends widget for the game bottom sheet. Mirrors RN
/// `components/nba/RecentTrendsWidget.tsx`. Collapsible card that lays out
/// 10 head-to-head metrics with directional color coding (green if a team's
/// number is better than the opponent's; lowerIsBetter inverts for the
/// defensive-rating row).
struct NBARecentTrendsWidget: View {
    let awayTeam: String
    let homeTeam: String
    let trends: NBAGameTrends?
    let isLoading: Bool

    /// Expansion is driven by the hosting `WidgetSection` header (title +
    /// chevron + card chrome live there), shared with the other pinned widgets.
    @Binding var expanded: Bool

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            if isLoading {
                ProgressView()
                    .padding(20)
                    .frame(maxWidth: .infinity)
            } else if expanded {
                content
            } else {
                Text("Tap to view recent head-to-head trends")
                    .font(.system(size: 13, weight: .medium))
                    .foregroundStyle(Color.appTextSecondary)
                    .frame(maxWidth: .infinity, alignment: .leading)
            }
        }
    }

    @ViewBuilder
    private var content: some View {
        if let trends {
            VStack(spacing: 8) {
                tableHeader
                ForEach(Array(metrics(trends).enumerated()), id: \.offset) { idx, metric in
                    metricRow(metric, index: idx)
                }
            }
            .padding(.horizontal, 12)
            .padding(.bottom, 12)
        } else {
            VStack(spacing: 12) {
                Image(systemName: "info.circle")
                    .font(.system(size: 32))
                    .foregroundStyle(Color.appTextSecondary)
                Text("No trend data available for this matchup")
                    .font(.system(size: 14))
                    .foregroundStyle(Color.appTextSecondary)
            }
            .frame(maxWidth: .infinity)
            .padding(24)
            .padding(.bottom, 12)
        }
    }

    @ViewBuilder
    private var tableHeader: some View {
        HStack {
            GameCardTeamAvatar(teamName: awayTeam, sport: "nba", size: 32)
                .frame(width: 70)
            Text("METRIC")
                .font(.system(size: 11, weight: .semibold))
                .tracking(0.5)
                .foregroundStyle(Color.appTextSecondary)
                .frame(maxWidth: .infinity)
            GameCardTeamAvatar(teamName: homeTeam, sport: "nba", size: 32)
                .frame(width: 70)
        }
        .padding(.bottom, 8)
        .overlay(Rectangle().frame(height: 1).foregroundStyle(Color.appBorder), alignment: .bottom)
    }

    private struct Metric {
        let name: String
        let awayValue: Double?
        let homeValue: Double?
        let format: (Double?) -> String
        let lowerIsBetter: Bool
        let noColor: Bool
    }

    private func metrics(_ t: NBAGameTrends) -> [Metric] {
        [
            Metric(name: "Overall Rating", awayValue: t.awayOvrRtg, homeValue: t.homeOvrRtg, format: Self.formatDecimal2, lowerIsBetter: false, noColor: false),
            Metric(name: "Consistency Rating", awayValue: t.awayConsistency, homeValue: t.homeConsistency, format: Self.formatDecimal2, lowerIsBetter: false, noColor: false),
            Metric(name: "Win Streak", awayValue: t.awayWinStreak, homeValue: t.homeWinStreak, format: Self.formatInt, lowerIsBetter: false, noColor: false),
            Metric(name: "ATS %", awayValue: t.awayAtsPct, homeValue: t.homeAtsPct, format: Self.formatPercent, lowerIsBetter: false, noColor: false),
            Metric(name: "ATS Streak", awayValue: t.awayAtsStreak, homeValue: t.homeAtsStreak, format: Self.formatInt, lowerIsBetter: false, noColor: false),
            Metric(name: "Last Game Score Margin", awayValue: t.awayLastMargin, homeValue: t.homeLastMargin, format: Self.formatDecimal1, lowerIsBetter: false, noColor: false),
            Metric(name: "Over/Under %", awayValue: t.awayOverPct, homeValue: t.homeOverPct, format: Self.formatPercent, lowerIsBetter: false, noColor: true),
            Metric(name: "Pace Trend (Last 3)", awayValue: t.awayAdjPacePregameL3Trend, homeValue: t.homeAdjPacePregameL3Trend, format: Self.formatDecimal2, lowerIsBetter: false, noColor: false),
            Metric(name: "Off. Rating Trend (L3)", awayValue: t.awayAdjOffRtgPregameL3Trend, homeValue: t.homeAdjOffRtgPregameL3Trend, format: Self.formatDecimal2, lowerIsBetter: false, noColor: false),
            Metric(name: "Def. Rating Trend (L3)", awayValue: t.awayAdjDefRtgPregameL3Trend, homeValue: t.homeAdjDefRtgPregameL3Trend, format: Self.formatDecimal2, lowerIsBetter: true, noColor: false)
        ]
    }

    @ViewBuilder
    private func metricRow(_ m: Metric, index: Int) -> some View {
        let awayColor = trendColor(my: m.awayValue, other: m.homeValue, lowerIsBetter: m.lowerIsBetter, noColor: m.noColor)
        let homeColor = trendColor(my: m.homeValue, other: m.awayValue, lowerIsBetter: m.lowerIsBetter, noColor: m.noColor)
        HStack {
            Text(m.format(m.awayValue))
                .font(.system(size: 13, weight: .bold))
                .foregroundStyle(awayColor)
                .frame(width: 70, alignment: .center)
            Text(m.name)
                .font(.system(size: 12, weight: .medium))
                .foregroundStyle(Color.appTextSecondary)
                .multilineTextAlignment(.center)
                .frame(maxWidth: .infinity)
            Text(m.format(m.homeValue))
                .font(.system(size: 13, weight: .bold))
                .foregroundStyle(homeColor)
                .frame(width: 70, alignment: .center)
        }
        .padding(.vertical, 8)
        .background(index.isMultiple(of: 2) ? Color.appTextMuted.opacity(0.04) : .clear)
    }

    private func trendColor(my: Double?, other: Double?, lowerIsBetter: Bool, noColor: Bool) -> Color {
        if noColor { return Color.appTextPrimary }
        guard let my, let other else { return Color.appTextPrimary }
        if lowerIsBetter {
            if my < other { return Color.appPrimary }
            if my > other { return Color.appAccentRed }
        } else {
            if my > other { return Color.appPrimary }
            if my < other { return Color.appAccentRed }
        }
        return Color.appTextPrimary
    }

    // MARK: - Format helpers

    private static func formatDecimal2(_ v: Double?) -> String {
        guard let v else { return "-" }
        return String(format: "%.2f", v)
    }

    private static func formatDecimal1(_ v: Double?) -> String {
        guard let v else { return "-" }
        return String(format: "%.1f", v)
    }

    private static func formatPercent(_ v: Double?) -> String {
        guard let v else { return "-" }
        return String(format: "%.1f%%", v * 100)
    }

    private static func formatInt(_ v: Double?) -> String {
        guard let v else { return "-" }
        return String(Int(v.rounded()))
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/nba/NBARecentTrendsWidget.kt
package com.wagerproof.app.features.nba

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.app.features.gamecards.GameCardTeamAvatar
import com.wagerproof.app.features.gamecards.NBATeams
import com.wagerproof.core.design.icons.AppIcon
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.NBAGameTrends
import java.util.Locale
import kotlin.math.roundToInt

/**
 * NBA recent-trends widget — port of iOS `NBARecentTrendsWidget`. Chromeless
 * collapsible 10-metric head-to-head table with directional color coding
 * (better = green/appPrimary, worse = red; `lowerIsBetter` inverts for the
 * defensive-rating trend row; O/U % is intentionally uncolored).
 */
@Composable
fun NBARecentTrendsWidget(
    awayTeam: String,
    homeTeam: String,
    trends: NBAGameTrends?,
    isLoading: Boolean,
    expanded: Boolean,
    modifier: Modifier = Modifier,
) {
    Column(modifier.fillMaxWidth()) {
        when {
            isLoading -> Box(Modifier.fillMaxWidth().padding(20.dp), contentAlignment = Alignment.Center) {
                CircularProgressIndicator(Modifier.size(22.dp), strokeWidth = 2.dp)
            }
            expanded -> Content(awayTeam, homeTeam, trends)
            else -> Text(
                "Tap to view recent head-to-head trends",
                color = AppColors.appTextSecondary,
                fontSize = 13.sp,
                fontWeight = FontWeight.Medium,
            )
        }
    }
}

@Composable
private fun Content(awayTeam: String, homeTeam: String, trends: NBAGameTrends?) {
    if (trends == null) {
        Column(
            Modifier.fillMaxWidth().padding(24.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(12.dp),
        ) {
            AppIcon.fromSystemName("info.circle")?.let {
                Icon(it.imageVector, null, tint = AppColors.appTextSecondary, modifier = Modifier.size(32.dp))
            }
            Text("No trend data available for this matchup", color = AppColors.appTextSecondary, fontSize = 14.sp)
        }
        return
    }
    Column(
        Modifier.fillMaxWidth().padding(horizontal = 12.dp).padding(bottom = 12.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Row(Modifier.fillMaxWidth().padding(bottom = 8.dp), verticalAlignment = Alignment.CenterVertically) {
            Box(Modifier.width(70.dp), contentAlignment = Alignment.Center) {
                GameCardTeamAvatar(sport = "nba", team = awayTeam, diameter = 32.dp, colors = NBATeams.colorPair(awayTeam))
            }
            Text(
                "METRIC",
                color = AppColors.appTextSecondary, fontSize = 11.sp, fontWeight = FontWeight.SemiBold,
                textAlign = TextAlign.Center, modifier = Modifier.weight(1f),
            )
            Box(Modifier.width(70.dp), contentAlignment = Alignment.Center) {
                GameCardTeamAvatar(sport = "nba", team = homeTeam, diameter = 32.dp, colors = NBATeams.colorPair(homeTeam))
            }
        }
        Box(Modifier.fillMaxWidth().height(1.dp).background(AppColors.appBorder))
        metrics(trends).forEachIndexed { index, metric ->
            MetricRow(metric, index)
        }
    }
}

private class Metric(
    val name: String,
    val awayValue: Double?,
    val homeValue: Double?,
    val format: (Double?) -> String,
    val lowerIsBetter: Boolean = false,
    val noColor: Boolean = false,
)

private fun metrics(t: NBAGameTrends): List<Metric> = listOf(
    Metric("Overall Rating", t.awayOvrRtg, t.homeOvrRtg, ::formatDecimal2),
    Metric("Consistency Rating", t.awayConsistency, t.homeConsistency, ::formatDecimal2),
    Metric("Win Streak", t.awayWinStreak, t.homeWinStreak, ::formatInt),
    Metric("ATS %", t.awayAtsPct, t.homeAtsPct, ::formatPercent),
    Metric("ATS Streak", t.awayAtsStreak, t.homeAtsStreak, ::formatInt),
    Metric("Last Game Score Margin", t.awayLastMargin, t.homeLastMargin, ::formatDecimal1),
    Metric("Over/Under %", t.awayOverPct, t.homeOverPct, ::formatPercent, noColor = true),
    Metric("Pace Trend (Last 3)", t.awayAdjPacePregameL3Trend, t.homeAdjPacePregameL3Trend, ::formatDecimal2),
    Metric("Off. Rating Trend (L3)", t.awayAdjOffRtgPregameL3Trend, t.homeAdjOffRtgPregameL3Trend, ::formatDecimal2),
    Metric("Def. Rating Trend (L3)", t.awayAdjDefRtgPregameL3Trend, t.homeAdjDefRtgPregameL3Trend, ::formatDecimal2, lowerIsBetter = true),
)

@Composable
private fun MetricRow(m: Metric, index: Int) {
    val awayColor = trendColor(m.awayValue, m.homeValue, m.lowerIsBetter, m.noColor)
    val homeColor = trendColor(m.homeValue, m.awayValue, m.lowerIsBetter, m.noColor)
    Row(
        Modifier
            .fillMaxWidth()
            .background(if (index % 2 == 0) AppColors.appTextMuted.copy(alpha = 0.04f) else Color.Transparent)
            .padding(vertical = 8.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Text(
            m.format(m.awayValue),
            color = awayColor, fontSize = 13.sp, fontWeight = FontWeight.Bold,
            textAlign = TextAlign.Center, modifier = Modifier.width(70.dp),
        )
        Text(
            m.name,
            color = AppColors.appTextSecondary, fontSize = 12.sp, fontWeight = FontWeight.Medium,
            textAlign = TextAlign.Center, modifier = Modifier.weight(1f),
        )
        Text(
            m.format(m.homeValue),
            color = homeColor, fontSize = 13.sp, fontWeight = FontWeight.Bold,
            textAlign = TextAlign.Center, modifier = Modifier.width(70.dp),
        )
    }
}

private fun trendColor(my: Double?, other: Double?, lowerIsBetter: Boolean, noColor: Boolean): Color {
    if (noColor) return AppColors.appTextPrimary
    if (my == null || other == null) return AppColors.appTextPrimary
    return if (lowerIsBetter) {
        when {
            my < other -> AppColors.appPrimary
            my > other -> AppColors.appAccentRed
            else -> AppColors.appTextPrimary
        }
    } else {
        when {
            my > other -> AppColors.appPrimary
            my < other -> AppColors.appAccentRed
            else -> AppColors.appTextPrimary
        }
    }
}

private fun formatDecimal2(v: Double?): String = v?.let { String.format(Locale.US, "%.2f", it) } ?: "-"
private fun formatDecimal1(v: Double?): String = v?.let { String.format(Locale.US, "%.1f", it) } ?: "-"
private fun formatPercent(v: Double?): String = v?.let { String.format(Locale.US, "%.1f%%", it * 100) } ?: "-"
private fun formatInt(v: Double?): String = v?.let { it.roundToInt().toString() } ?: "-"
